salt/data-pipeline/bigquery-views.sls

```diff
@@ -11,10 +11,15 @@ re-materialise views daily:
     cron.absent:
         - identifier: materialize-views-daily
 
+
 re-materialise views:
     cron.present:
         - user: {{ pillar.elife.deploy_user.username }}
+        {% if pillar.elife.env == "prod" %}
```

Comment:
I think we usually put the environment specific configuration in the config that defines the properties, for each environment. Although `--disable-view-name-mapping` doesn't lend itself to it, so it's probably be okay this way. Will refer to @giorgiosironi for advice / thoughts.

Reply:
thought of doing that , but the `pillar.elife.env`  is managed outside of this formula, and that is why i deferred to including the text directly in the cron job configuration

Reply:
1. Create `pillar.data_pipeline.bigquery_views.materialize_arguments` in https://github.com/elifesciences/data-pipeline-formula/blob/master/salt/pillar/data-pipeline.sls#L23
1. Add empty string value to https://github.com/elifesciences/builder-configuration/blob/master/pillar/data-pipeline-public.sls#L3
1. Add `--disable-view-name-mapping` value to https://github.com/elifesciences/builder-configuration/blob/master/pillar/environment-prod-public.sls (should be in the same PR, needs to add the parent keys in the YAML)
1. Change to use `{{pillar.data_pipeline.bigquery_views.materialize_arguments }}` removing the if

This way we can tune this by environment without changing the formula but only the configuration.

Separately: why `staging` shouldn't have the same behavior in principle?

Reply:
I can help with parts of this, will create the `builder-configuration` change.

Reply:
> thought of doing that , but the `pillar.elife.env` is managed outside of this formula, and that is why i deferred to including the text directly in the cron job configuration

@de-code : what do you think about having this? implication for the "view-mapping" functionality will be having `[ deployment_env_name ]_[ mapped_dataset_name ] ` as the eventual dataset name for mapped views ? if this is ok with you, should we implement this now, and leave this formula unmodified, or should we temporarily modify this formula to resolve the civi issue and then later implement this dataset name change functionality?

Reply:
> @de-code : what do you think about having this? implication for the "view-mapping" functionality will be having `[ deployment_env_name ]_[ mapped_dataset_name ] ` as the eventual dataset name for mapped views ? if this is ok with you, should we implement this now, and leave this formula unmodified, or should we temporarily modify this formula to resolve the civi issue and then later implement this dataset name change functionality?

I think in principal, doing the mapping for every deployment environment seems to make sense. That would however also affect the dataset name that the Civi intergration currently uses.

I think for now I would recommend doing the minimal to resolve the Civi issue and change the implementation later. (The only change I would maybe consider now is changing the default to avoid the exact same issue)